x-pack/plugins/enterprise_search/server/lib/enterprise_search_config.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { readFileSync } from 'fs';
+
+import http from 'http';
+import https from 'https';
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+// -----------------------------------------------------------------------------
+export const configSchema = schema.object({
+  host: schema.maybe(schema.string()),
+  enabled: schema.boolean({ defaultValue: true }),
+  accessCheckTimeout: schema.number({ defaultValue: 5000 }),
+  accessCheckTimeoutWarning: schema.number({ defaultValue: 300 }),
+  ssl: schema.object({
+    certificateAuthorities: schema.maybe(
+      schema.oneOf([schema.arrayOf(schema.string()), schema.string()])
+    ),
+    rejectUnauthorized: schema.boolean({ defaultValue: false }),
+  }),
+});
+
+type EnterpriseSearchConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<EnterpriseSearchConfigType> = {
+  schema: configSchema,
+  exposeToBrowser: {
+    host: true,
+  },
+};
```

Comment:
Actually, on 2nd thought, after re-reading this - I don't think we should move this to the `lib/enterprise_search_config.ts` file. I'd strongly push to leave the `config`/`configSchema` exports within `server/index.ts`:

- It provides a single easy-to-glance location where people can come in to our plugin and quickly see what our config yaml options are, as well as which are available in the UI
- It matches other Kibana plugins that leave their config schema in their `server/index.ts`

I think this class alone does a great job of documenting the config options more in detail, and it doesn't need this additionally at the top of the file.